### ISSUE TITLE
Replies still have mangled links in them

### ISSUE THREAD

**Issue by nwhitehorn** [cphyc/thunderbird_remove_safelinks#3]
I'm not sure how to fix this -- I guess there needs to be a matching composescript hook -- but thought I would put it here for reference.

**Comment by cphyc** [comment]
This would require injecting the decoders in the compose window (see https://thunderbird-webextensions.readthedocs.io/en/latest/composeScripts.html). This should not be too hard.

**Comment by cphyc** [comment]
Let me detail a bit more: the current extension only changes what is _displayed_ in the messages. You can see it by inspecting the source of the message (Ctrl+U) that still contains the original message and links.

In order to include the unmingled links in replies, one would have to apply the same filtering to the compose window by injecting a `ComposeScript`, see https://thunderbird-webextensions.readthedocs.io/en/latest/composeScripts.html.

In principle, this should be super simple to implement so any PR is welcome!

**Comment by jan-kiszka** [comment]
Is anyone working on this already? I was just looking into composeScripts for "Signature Switch" and could try to transfer this knowledge.

**Comment by cphyc** [comment]
@jan-kiszka you can have a look at #4 , which just needs to be rebased again the current main branch.

**Comment by jan-kiszka** [comment]
`npm run build` fails in all colors here. So I can't reproduce the package, even from main.

Seems `npm install` pulls instable versions of its dependencies, e.g.

```
thunderbird_remove_safelinks/node_modules/update-notifier/index.js:58
                        } catch {
                                ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (thunderbird_remove_safelinks/node_modules/web-ext/dist/web-ext.js:1:48115)
```

FWIW, pushed the rebase to https://github.com/jan-kiszka/thunderbird_remove_safelinks/commits/queue

**Comment by cphyc** [comment]
@jan-kiszka, it works well on my side (with node v12.16.1 and npm 6.13.4). Which version are you using?

**Comment by jan-kiszka** [comment]
node8 (SUSE). Stone-age in nodejs terms, I know very well. My gcc is newer, I use it more frequently.

Fortunately, all that npm stuff isn't needed for packaging also here, just a simple zip command. That's what I use in my addons, in a makefile rule.

**Comment by nwhitehorn** [comment]
Fix merged in PR #8
